bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/WorkerDataChannel.java

```diff
@@ -65,7 +65,7 @@ public synchronized boolean putData(BRefType data, Strand waitingCtx, int retReg
             waitingCtx.setError(this.panic);
             BVM.handleError(waitingCtx);
             this.panic = null;
-            return false;
+            return true;
```

Comment:
Isn't BVM.handleError continue the execution? If that so we should return false here.

Reply:
no it is not, it just pops the frames if the current frame cannot handle the error.